File: Leetcode/cpp/sol2255.cpp

//https://leetcode.com/problems/count-prefixes-of-a-given-string/

class Solution {
public:
    bool startswith(string word, string prefix){
        /*
        Just a custom fuction to check the start of the word and see if it matches with the prefix
        better use compare() or substr()
        */
        if(word.size() < prefix.size()){
            return false;
        }

        for(int i = 0; i < prefix.size(); i++){
            if (word[i] != prefix[i]){
                return false;
            }
        }
        return true;
    }

    int countPrefixes(vector<string>& words, string s) {
        int count = 0;
        int word_length = s.size();
        for(string prefix: words){
            cout << s.substr(0, prefix.size())<<endl;
            if (word_length < prefix.size(), s.compare(0, prefix.length(), prefix) == 0){
                count++;
            }
        }

        return count;
    }
};

File: Leetcode/python/sol2255.py

# https://leetcode.com/problems/count-prefixes-of-a-given-string/solutions/7114756/startwith-by-antoniojsp-0nav/

class Solution:
    def countPrefixes(self, words: List[str], s: str) -> int:
        count = 0
        for i in words:
            if s.startswith(i):
                count+=1
        return count